src/main/java/com/courtcanva/ccfranchise/services/FranchiseeService.java

```diff
@@ -0,0 +1,41 @@
+package com.courtcanva.ccfranchise.services;
+
+import com.courtcanva.ccfranchise.dtos.FranchiseeAndStaffInfoDto;
+import com.courtcanva.ccfranchise.dtos.ResponseDto;
+import com.courtcanva.ccfranchise.exceptions.ResourceNotFoundException;
+import com.courtcanva.ccfranchise.mappers.FranchiseeMapper;
+import com.courtcanva.ccfranchise.models.Franchisee;
+import com.courtcanva.ccfranchise.repositories.FranchiseeRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class FranchiseeService {
+    private final FranchiseeRepository franchiseeRepository;
+
+    private final FranchiseeMapper franchiseeMapper;
+
+    private final StaffService staffService;
+
+    @Transactional
+    public ResponseDto createFranchiseeAndStaff(FranchiseeAndStaffInfoDto franchiseeAndStaffInfoDto) {
+
+        franchiseeRepository.save(franchiseeMapper
+                .toFranchiseeEntity(franchiseeAndStaffInfoDto));
+
+        staffService.createStaff(franchiseeAndStaffInfoDto.getStaff());
+
+        return ResponseDto.builder()
+                .responseCode("201")
+                .responseMessage("create new Franchisee with new Staff")
+                .build();
+    }
+
+    public Franchisee findFranchiseeByABN(String abn) {
+
+        return franchiseeRepository.findByAbn(abn)
+                .orElseThrow(() -> new ResourceNotFoundException("Franchisee", "abn", abn));
```

Comment:
As you can see, if a newcomer is reading the code, he has to jump to `ResourceNotFoundException` first to know how it works under the hood. For making the code more cohesive, we can just define the error message here instead. Also, please add logs for documenting this error when it occurs.

Reply:
thx lao. now I pass a message to ResoureceNotFoundException which extends RuntimeException